sdm.R

```diff
@@ -0,0 +1,147 @@
+# From forecasting-bbs-R.ipynb --------------------------------------------
+sapply(dir("R", full.names = TRUE), source)
+start_yr <- 1982
+end_yr <- 2013
+min_num_yrs <- 25
+
+# start SDM ---------------------------------------------------------------
+library(gbm)
+library(parallel)
+library(dplyr)
+library(tidyr)
+
+set.seed(0)
+mc.cores = 16
+n_test_years = 5
+interaction.depth = 8
+n.trees = 1E4
+
+train_years = start_yr:(end_yr - n_test_years)
+test_years = (end_yr - n_test_years + 1):end_yr
+
+# Parameters for spatial cross-validation
+n_clusters = 100
+n_folds = 10
+stopifnot(n_clusters %% n_folds == 0)
+
+# Load data ---------------------------------------------------------------
+occurrences = get_bbs_data(start_yr = 1982, end_yr = 2013, min_num_yrs = 25)
+
+env = get_env_data() %>%
+  filter_ts(start_yr, end_yr, min_num_yrs) %>%
+  inner_join(distinct(occurrences, site_id, lat, long, year),
```

Comment:
This `inner_join(distinct())` puts the `occurrence` and `species_id` columns in the `env` data frame, which ends up getting used in the models. That definitely doesn't look intentional unless your going for some sort of pseudo-joint sdm. =)